go/ai/retriever.go

```diff
@@ -49,14 +49,32 @@ type RetrieverResponse struct {
 	Documents []*Document `json:"documents"`
 }
 
-// RegisterRetriever registers the actions for a specific retriever.
-func RegisterRetriever(name string, retriever Retriever) {
-	core.RegisterAction(name,
-		core.NewAction(name, core.ActionTypeRetriever, nil, retriever.Retrieve))
-
-	core.RegisterAction(name,
-		core.NewAction(name, core.ActionTypeIndexer, nil, func(ctx context.Context, req *IndexerRequest) (struct{}, error) {
-			err := retriever.Index(ctx, req)
-			return struct{}{}, err
-		}))
+// DefineRetriever takes index and retrieve functions that access a vector DB
+// and returns a new Retriever that wraps them in registered actions.
+func DefineRetriever(
+	name string,
+	index func(context.Context, *IndexerRequest) error,
+	retrieve func(context.Context, *RetrieverRequest) (*RetrieverResponse, error),
+) Retriever {
+	ia := core.NewAction(name, core.ActionTypeIndexer, nil, func(ctx context.Context, req *IndexerRequest) (struct{}, error) {
+		return struct{}{}, index(ctx, req)
+	})
+	core.RegisterAction(name, ia)
```

Comment:
Suggestion: add `core.DefineAction` function that does this -- `NewAction` + `RegisterAction`. 

Ex: https://github.com/firebase/genkit/blob/5dd2eaebd912915033e83006ee90cb44a67d94f1/js/core/src/action.ts#L209